model/job.go

```diff
@@ -17,21 +17,46 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type jobLinkProvider struct {
+	role *Role
+	job  *Job
+}
+
+// JobLinkProvides describes the BOSH links a job provides
+type JobLinkProvides struct {
+	Name       string
+	Role       *Role
+	Job        *Job
+	Type       string   `json:"type"`
+	Properties []string `json:"properties"`
+	providers  []jobLinkProvider
```

Comment:
Why is providers a list? It looks like it is either empty or just one element. Since the members of the type are pointers, you could just use `nil` to indicate that the list is empty?

Also, how is `providers.Role` different from `Role` (and `providers.Job different from `Job`)?